sycl/test-e2e/Graph/in_order_mixed_use.cpp

```diff
@@ -0,0 +1,59 @@
+// REQUIRES: cuda || level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests submitting an host kernel to an in-order queue before recording
+// commands from it.
+
+#include "graph_common.hpp"
+
+int main() {
+  using T = int;
+
+  queue Queue{{sycl::ext::intel::property::queue::no_immediate_command_list{},
+               sycl::property::queue::in_order{}}};
+  T *TestData = sycl::malloc_shared<T>(Size, Queue);
+
+  ext::oneapi::experimental::command_graph Graph{Queue.get_context(),
+                                                 Queue.get_device()};
+  Queue.submit([&](handler &CGH) {
+    CGH.host_task([=]() {
+      for (size_t i = 0; i < Size; i++) {
+        TestData[i] = static_cast<T>(i);
+      }
+    });
+  });
+
+  Graph.begin_recording(Queue);
+
+  auto GraphEvent = Queue.submit([&](handler &CGH) {
+    CGH.single_task<class TestKernel2>([=]() {
+      for (size_t i = 0; i < Size; i++) {
+        TestData[i] += static_cast<T>(i);
+      }
+    });
+  });
+
+  Graph.end_recording(Queue);
+
+  auto GraphExec = Graph.finalize();
+  Queue.submit([&](handler &CGH) { CGH.ext_oneapi_graph(GraphExec); });
+
+  Queue.submit([&](handler &CGH) {
+    CGH.single_task<class TestKernel3>([=]() {
```

Comment:
Should this be a host task as well to verify the dependency between the graph event and this kernel?

Reply:
In this test, I would like to test mix tasks host/device.
Fine checking of dependencies is carried out by unitests.